samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnContainerRunner.java

```diff
@@ -110,6 +111,7 @@ public void runContainer(String samzaContainerId, Container container, CommandBu
     log.info("Container ID {} using command {}", samzaContainerId, command);
 
     Map<String, String> env = getEscapedEnvironmentVariablesMap(cmdBuilder);
+    env.put(ShellCommandConfig.ENV_EXECUTION_ENV_CONTAINER_ID(), Util.envVarEscape(container.getId().toString()));
```

Comment:
The Yarn container Id is only available to be set in two places: here and in YarnClusterResourceManager. Unless I make changes to ShellCommandBuilder to be able to add new configs (via a setter??) I don't this this environment var can be set elsewhere